### ISSUE TITLE
Transform request body

### ISSUE THREAD

**Issue by cdtut** [honojs/hono#2467]
### What is the feature you are proposing?

Routes get request body with functions like `await c.req.json()`. In middleware you should be able to intercept and change the value the route gets with something like `c.req.setJson(someTransformMethod(await c.req.json()))`.

**Comment by NicoPlyley** [comment]
If you would like to do that in the meantime, you can use `c.set() / c.get()` and pass the updated data to your route that way instead

**Comment by yusukebe** [comment]
As @NicoPlyley said, we can use `c.set() / c.get()`. Closing this.

**Comment by cdtut** [comment]
@yusukebe `c.get()` doesn't have same benefit. If some routes body need to be transformed and others don't it is inconsistent. Some will have to use `c.get().json` and others use `c.req.json()` which is confusing and bring mistakes.

If possible to transform a body before getting to route all route can use `c.req.json()` and know it is right data.

**Comment by yusukebe** [comment]
@cdtut 

This way is not recommended because this API is for internal usage, but you can make it with the code:

```ts
app.use('*', async (c, next) => {
  const data = await c.req.json()
  //
  c.req.bodyCache.json = 'transformeddata'
  await next()
})

app.post('/', async (c) => {
  const data = await c.req.json()
  return c.json(data)
})
```

**Comment by cdtut** [comment]
Can internal `c.req.bodyCache.json = data` have external API like `c.req.setJson(data)` so can be used reliably? For all body requests not just json.

**Comment by yusukebe** [comment]
> Can internal `c.req.bodyCache.json = data` have external API like `c.req.setJson(data)` so can be used reliably?

No. Basically, the Request body should be immutable, and we don't want to add more methods to the `c.req`.

**Comment by maliknajjar** [comment]
Since the author of the library intends to keep the request body immutable, for @cdtut or anyone else who needs a consistent way to interact with and modify the body, here's a solution:

You can create a root middleware to handle the request body. Capture the value using `await c.req.json()` and store it in a variable like `body`.

```js
const app = new Hono();

app.use(async (c: Context, next: Next) => {
    const body = await c.req.json();
    c.set('body', body);
    await next();
});
```

From this point onward, you can consistently access and interact with the body throughout the application using the stored variable. Retrieve the body with `c.get('body')`, and update it similarly with `c.set('body', newBody)`.
